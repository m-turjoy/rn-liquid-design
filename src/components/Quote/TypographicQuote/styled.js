import styled from 'styled-components';
import {
  width,
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight,
  color,
} from 'styled-system';

const QuoteWrapper = styled.View`
  ${width};
  flexdirection: column;
  justify-content: center;
  align-items: flex-start;
`;

const AuthorWrapper = styled.Text`
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${lineHeight}
  ${color}
  margin-top: 10
`;
const PhraseWrapper = styled.Text`
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${lineHeight}
  ${color}
`;

export { QuoteWrapper, AuthorWrapper, PhraseWrapper };
