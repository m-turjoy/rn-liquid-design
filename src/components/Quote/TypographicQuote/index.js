import React from "react";
import { bool, number, string } from "prop-types";
import { fonts, colors } from "../../../config";
import { QuoteWrapper, PhraseWrapper, AuthorWrapper } from "./styled";

const TypographicQuote = ({
	author,
	big,
	small,
	quotation,
	width,
	fontFamily,
	fontSize,
	fontWeight,
	lineHeight,
	color,
	authorFontFamily,
	authorFontSize,
	authorFontWeight,
	authorLineHeight,
	authorColor,
}) => (
	<QuoteWrapper width={width}>
		<PhraseWrapper
			fontFamily={fontFamily || fonts.Black}
			fontSize={big ? 32 : small ? 22 : fontSize}
			fontWeight={fontWeight || null}
			lineHeight={big ? 40 : small ? 27.5 : lineHeight}
			color={color || colors.richBlackDefault}
		>
			{`»${quotation}«`}
		</PhraseWrapper>
		<AuthorWrapper
			fontFamily={authorFontFamily || fonts.Regular}
			fontSize={authorFontSize || 12}
			fontWeight={authorFontWeight || null}
			lineHeight={authorLineHeight || 15}
			color={authorColor || colors.vibrantCyanDefault}
		>
			{`– ${author}`}
		</AuthorWrapper>
	</QuoteWrapper>
);

TypographicQuote.propTypes = {
	quotation: string.isRequired,
	author: string.isRequired,
	big: bool,
	small: bool,
	width: number,
	fontFamily: string,
	fontSize: number,
	fontWeight: number,
	lineHeight: number,
	color: string,
	authorFontFamily: string,
	authorFontSize: number,
	authorFontWeight: number,
	authorLineHeight: number,
	authorColor: string,
};

TypographicQuote.defaultProps = {
	big: false,
	width: 250,
};

export default TypographicQuote;
