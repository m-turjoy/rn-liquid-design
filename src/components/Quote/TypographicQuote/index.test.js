import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import TypographicQuote from './index'

configure({ adapter: new Adapter() })
describe('TypographicQuote component', () => {
  describe('Renders correctly', () => {
    test('it renders big TypographicQuote', () => {
      const tree = shallow(
        <TypographicQuote
          quotation='Test'
          author='Test'
          big
        />,
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
