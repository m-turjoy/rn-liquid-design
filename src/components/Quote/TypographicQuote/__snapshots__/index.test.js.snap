// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypographicQuote component Renders correctly it renders big TypographicQuote 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TypographicQuote
    author="Test"
    big={true}
    quotation="Test"
    width={250}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Text)
          color="#1b1b25"
          fontFamily="Lato-Black"
          fontSize={32}
          fontWeight={null}
          lineHeight={40}
        >
          »Test«
        </Styled(Text)>,
        <Styled(Text)
          color="#2dbecd"
          fontFamily="Lato-Regular"
          fontSize={12}
          fontWeight={null}
          lineHeight={15}
        >
          – Test
        </Styled(Text)>,
      ],
      "width": 250,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "»Test«",
          "color": "#1b1b25",
          "fontFamily": "Lato-Black",
          "fontSize": 32,
          "fontWeight": null,
          "lineHeight": 40,
        },
        "ref": null,
        "rendered": "»Test«",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "– Test",
          "color": "#2dbecd",
          "fontFamily": "Lato-Regular",
          "fontSize": 12,
          "fontWeight": null,
          "lineHeight": 15,
        },
        "ref": null,
        "rendered": "– Test",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Text)
            color="#1b1b25"
            fontFamily="Lato-Black"
            fontSize={32}
            fontWeight={null}
            lineHeight={40}
          >
            »Test«
          </Styled(Text)>,
          <Styled(Text)
            color="#2dbecd"
            fontFamily="Lato-Regular"
            fontSize={12}
            fontWeight={null}
            lineHeight={15}
          >
            – Test
          </Styled(Text)>,
        ],
        "width": 250,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "»Test«",
            "color": "#1b1b25",
            "fontFamily": "Lato-Black",
            "fontSize": 32,
            "fontWeight": null,
            "lineHeight": 40,
          },
          "ref": null,
          "rendered": "»Test«",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "– Test",
            "color": "#2dbecd",
            "fontFamily": "Lato-Regular",
            "fontSize": 12,
            "fontWeight": null,
            "lineHeight": 15,
          },
          "ref": null,
          "rendered": "– Test",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
